__tests__/app.test.js

```diff
@@ -138,4 +138,52 @@ describe('GET/api/articles/:article_id/comments', ()=> {
       expect(body.msg).toBe('bad request')
     })
   });
-})
\ No newline at end of file
+})
+
+describe ('POST /api/articles/:article_id/comments', () => {
+  test('201 - new comment added to table', () => {
+    const newComment = {
+      username: 'butter_bridge',
+      body: "this article is a big pile of poo"
+    }
+    return request(app)
+    .post('/api/articles/1/comments')
+    .send(newComment)
+    .expect(201)
+    .then(({body}) => {
+      expect(body.comment).toEqual({
+        comment_id: 19,
+        article_id: 1, 
+        created_at: expect.any(String),
+        votes: expect.any(Number),
+        author: 'butter_bridge',
+        body: "this article is a big pile of poo"
+      })
+    })
+  });
+  test('404 - article id valid but non-existent', () => {
+    const newComment = {
+      username: 'butter_bridge',
+      body: "this article is a big pile of poo"}
+      return request(app)
+      .post('/api/articles/2050/comments')
+    .send(newComment)
+    .expect(404)
+    .then(({body}) => {
+      expect(body.msg).toBe('article not found')
+    })
+    })
+    test('400 - not a valid article id', () => {
+      const newComment = {
+        username: 'butter_bridge',
+        body: "this article is a big pile of poo"}
+        return request(app)
+        .post('/api/articles/invalid/comments')
+      .send(newComment)
+      .expect(400)
+      .then(({body}) => {
+        expect(body.msg).toBe('bad request')
+      })
+    })
```

Comment:
We are missing a few sad path test here. 1. It's possible for username to be invalid. 2 It's possible for username to be misspelled. 3. It's possible for body to be misspelled